src/ScheduledTasks/Client/ScheduledTaskClientImpl.cs

```diff
@@ -0,0 +1,186 @@
+        return Pageable.Create(async (continuationToken, pageSize, cancellation) =>
+        {
+            try
+            {
+                // TODO: map to entity query last modified from/to filters
+                EntityQuery query = new EntityQuery
+                {
+                    InstanceIdStartsWith = filter?.ScheduleIdPrefix ?? nameof(Schedule),
+                    IncludeState = true,
+                    PageSize = filter?.PageSize ?? ScheduleQuery.DefaultPageSize,
+                    ContinuationToken = continuationToken,
+                };
+
+                // Get one page of entities
+                IAsyncEnumerable<Page<EntityMetadata<ScheduleState>>> entityPages =
+                    this.durableTaskClient.Entities.GetAllEntitiesAsync<ScheduleState>(query).AsPages();
+
+                await foreach (Page<EntityMetadata<ScheduleState>> entityPage in entityPages)
+                {
+                    List<ScheduleDescription> schedules = entityPage.Values
+                        .Where(metadata =>
+                            (!filter?.Status.HasValue ?? true || metadata.State.Status == filter.Status.Value) &&
+                            (filter?.CreatedFrom.HasValue != true || metadata.State.ScheduleCreatedAt > filter.CreatedFrom) &&
+                            (filter?.CreatedTo.HasValue != true || metadata.State.ScheduleCreatedAt < filter.CreatedTo))
+                        .Select(metadata =>
+                        {
+                            ScheduleState state = metadata.State;
+                            ScheduleConfiguration config = state.ScheduleConfiguration!;
+                            return new ScheduleDescription
+                            {
+                                ScheduleId = metadata.Id.Key,
+                                OrchestrationName = config.OrchestrationName,
+                                OrchestrationInput = config.OrchestrationInput,
+                                OrchestrationInstanceId = config.OrchestrationInstanceId,
+                                StartAt = config.StartAt,
+                                EndAt = config.EndAt,
+                                Interval = config.Interval,
+                                StartImmediatelyIfLate = config.StartImmediatelyIfLate,
+                                Status = state.Status,
+                                ExecutionToken = state.ExecutionToken,
+                                LastRunAt = state.LastRunAt,
+                                NextRunAt = state.NextRunAt,
+                            };
+                        })
+                        .ToList();
+
+                    return new Page<ScheduleDescription>(schedules, entityPage.ContinuationToken);
+                }
+
+                // Return empty page if no results
+                return new Page<ScheduleDescription>(new List<ScheduleDescription>(), null);
+            }
+            catch (OperationCanceledException e)
```

Comment:
See earlier comment about handling cancellation exceptions (I think you may have missed this one when refactoring all the other ones).